custom-routes-vis/src/main/webapp/script.js

```diff
@@ -143,9 +156,140 @@ function hideMarker(markerName) {
+      + '&endpoint=' + serviceEndpoint + '&apiKey=' + apiKey)
+      .then(response => response.json()).then(directions => {
+        // Log the response status from the Directions API.
+        // TODO(chenyuz): find equivalence for Routes Preferred API.
+        console.log(directions.status);
+
+        let routes = directions.routes;
+        console.log('num routes:', routes.length);
+
+        for (let routeNum = 0; routeNum < routes.length; routeNum++) {
+          createRoutePolyline(routeNum, routes[routeNum], 
+              serviceEndpoint === 'directions');
+        }
+      });
+}
+
+/**
+ * Creates a polyline for a route on the map.
+ * @param {num} routeNum The index of the route in the order as it is returned
+ * from the API.
+ * @param {!JSON} routeJson JSON object containing all information of the target
+ * route.
+ * @param {bool} directionsApi Whether the route is obtained by the Directions
+ * API; if false, assume it is obtained by the Routes Preferred API.
+ */
+function createRoutePolyline(routeNum, routeJson, directionsApi) {
+  let routeLegs = routeJson.legs;
+  let routeCoordinates = [];
+
+  // Total duration of route in seconds.
+  let totalDuration = directionsApi ? 0 : routeJson.duration;
+  // Total distance of route in meters.
+  let totalDistance = directionsApi ? 0 : routeJson.distanceMeters;
+  // Note: Routes Preferred has duration and distanceMeters attributes for each
+  // route, but Directions only has those for each leg of the route.
+
+  for (let i = 0; i < routeLegs.length; i++) {
+    let legSteps = routeLegs[i].steps;
+    console.log('num steps', legSteps.length);
+
+    for (let j = 0; j < legSteps.length; j++) {
+      if (directionsApi) {
+        routeCoordinates.push(legSteps[j].start_location);
+      } else {
+        routeCoordinates.push(legSteps[j].startLocation.LatLng);
+      }
+    }
+
+    // Add the end location to the coordinates array.
+    if (i == routeLegs.length - 1) {
+      if (directionsApi) {
+        routeCoordinates.push(legSteps[legSteps.length - 1].end_location);
+      } else {
+        routeCoordinates.push(legSteps[legSteps.length - 1].endLocation.LatLng);
+      }
+    }
+
+    if (directionsApi) {
+      totalDuration += parseInt(routeLegs[i].duration.value);
+      totalDistance += parseInt(routeLegs[i].distance.value);
+    }
+  }
+
+  createRouteFromCoordinates(routeCoordinates, routeNum, totalDuration,
+      totalDistance);
+}
+
+/**
+ * Creates the route on the map and stores it in the routes array.
+ */
+function createRouteFromCoordinates(routeCoordinates, routeNum, totalDuration, 
+    totalDistance) {
+  let route = new google.maps.Polyline({
+    path: routeCoordinates,
+    geodesic: true,
+    strokeColor: routeColors[routeNum % 3],
+    strokeOpacity: 0.5,
+    strokeWeight: 6,
+  });
+  route.setMap(map);
+  displayedRoutes.push(route);
+
+  // Select the first route as default.
+  if (routeNum == 0) {
```

Comment:
Change to ===.

Reply:
Is there an advantage for === over ==?

Reply:
https://google.github.io/styleguide/jsguide.html#features-equality-checks